lib/core/mergeConfig.js

```diff
@@ -56,14 +56,19 @@ module.exports = function mergeConfig(config1, config2) {
     .concat(defaultToConfig2Keys);
 
   var otherKeys = Object
-    .keys(config2)
+    .keys(config1)
+    .concat(Object.keys(config2))
     .filter(function filterAxiosKeys(key) {
       return axiosKeys.indexOf(key) === -1;
     });
 
-  utils.forEach(otherKeys, function otherKeysDefaultToConfig2(prop) {
-    if (typeof config2[prop] !== 'undefined') {
+  utils.forEach(otherKeys, function mergeDeepProperties(prop) {
```

Comment:
Both line34 and line65 use the "mergeDeepProperties" function, I personally think that this function should be extracted